packages/vue-apollo-composable/src/useResult.ts

```diff
@@ -16,9 +16,9 @@ export type UseResultReturn<T> = Readonly<Ref<Readonly<T>>>
  * @param  {Ref<TResult>} result A `result` returned from `useQuery` to resolve.
  * @returns Readonly ref with `undefined` or the resolved `result`.
  */
-export function useResult<TResult, TResultKey extends keyof TResult = keyof TResult> (
+export function useResult<TResult, TResultKey extends keyof NonNullable<TResult> = keyof NonNullable<TResult>> (
```

Comment:
Why the `NonNulalble` added here?

Reply:
`useResult` returns either the first key of `result` if there is one, or `result` itself. If `TResult` is an object, typescript should extract that specific property. The problem is that typescript cannot resolve properties of `TResult | undefined`, it can only access properties on `TResult`.

`useResult-types.test.ts` will fail otherwise